rust_src/src/buffers.rs

```diff
@@ -135,6 +136,31 @@ impl LispBufferRef {
 
         unsafe { *(self.beg_addr().offset(offset + n - self.beg_byte())) as u8 }
     }
+
+    #[inline]
+    pub fn fetch_multibyte_char(&self, n: ptrdiff_t) -> c_int {
```

Comment:
It's a shame we don't have a type for characters, but that would require changes to multibyte.rs.